subsys/shell/shell_ops.c

```diff
@@ -97,6 +98,51 @@ void shell_op_cursor_move(const struct shell *shell, s16_t val)
 	shell->ctx->cmd_buff_pos = new_pos;
 }
 
+static u16_t shift_calc(const char *str, u16_t pos, u16_t len, s16_t sign)
+{
+	bool found = false;
+	u16_t ret = 0;
+	u16_t idx;
+
+	while (1) {
+		idx = pos + ret * sign;
+		if (((idx == 0) && (sign < 0)) ||
+		    ((idx == len) && (sign > 0))) {
+			break;
+		}
+		if (isalnum((int)str[idx]) != 0) {
+			found = true;
+		} else {
+			if (found) {
+				break;
+			}
+		}
+		ret++;
+	}
+
+	return ret;
+}
+
+void shell_op_cursor_word_move(const struct shell *shell, s16_t val)
+{
+	s16_t shift;
+	s16_t sign;
+
+	if (val < 0) {
```

Comment:
Prototype does not match body: Regardless of the val, the movement will be the same in each direction.
What happen with val = 0?